### PR TITLE
integrate to cms ecosystem page

### PR DESCRIPTION
**Pull request by Slavik191**
## Description

Integrate to cms ecosystem page

### CODE REVIEW

File: features/cms/customWidgets/widgets/svg/svg.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    useEffect(() => {
+      const uppyInstance = new Uppy({
+        restrictions: {
+          allowedFileTypes: ["image/svg+xml"],
+        },
+        autoProceed: false,
+      });
+
+      uppyInstance.on("file-added", (file) => {
```
Comment (Jabher) [inline features/cms/customWidgets/widgets/svg/svg.tsx:69]: can be async fn

File: features/cms/customWidgets/widgets/svg/svg.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+import { Wrapped, ImageWrapper, Image, Button } from "./svgStyles";
+
+const readFile = (file: UppyFile): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader();
+    reader.onload = (event: ProgressEvent<FileReader>) =>
+      event.target && typeof event.target.result === "string"
+        ? resolve(event.target.result)
+        : "";
```
[the remark this answers is not in the document]
Comment (Slavik191) [reply features/cms/customWidgets/widgets/svg/svg.tsx]: done

File: features/cms/customWidgets/widgets/svg/svg.tsx (hunk 1 above)
Comment (Slavik191) [reply features/cms/customWidgets/widgets/svg/svg.tsx]: done

File: features/cms/customWidgets/widgets/info/info.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+            const descriptionData = infoBlock.get("description");
+            const linksData = infoBlock.get("links");
+
+            const description = descriptionData ? (
+              <Text>{descriptionData ?? ""}</Text>
+            ) : null;
+
+            const links: ReactNode[] = [];
+            (linksData || []).forEach((link: any) => {
```
Comment (Jabher) [inline features/cms/customWidgets/widgets/info/info.tsx:39]:
```suggestion
            linksData?.forEach((link: any) => {
```

File: features/cms/customWidgets/widgets/uuid/uuid.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+const UuidPreview = ({
+  value,
+  field,
+}: CmsWidgetPreviewProps<string | null>) => {
+  let hide = field.get("hide");
+  let ref = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if (hide) hideElement(ref);
+  }, []);
```
[the remark this answers is not in the document]
Comment (Slavik191) [reply features/cms/customWidgets/widgets/uuid/uuid.tsx:72]: Why?
This value is specified in the configuration and it is not dynamic

File: features/cms/customWidgets/widgets/info/info.tsx (hunk 3 above)
Comment (Slavik191) [reply features/cms/customWidgets/widgets/info/info.tsx]: done

File: features/cms/customWidgets/widgets/svg/svg.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import { Wrapped, ImageWrapper, Image, Button } from "./svgStyles";
+
+const readFile = (file: UppyFile): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader();
+    reader.onload = (event: ProgressEvent<FileReader>) =>
+      event.target && typeof event.target.result === "string"
+        ? resolve(event.target.result)
+        : reject("svg loading error");
+    reader.onerror = (err) => reject(err);
```
[the remark this answers is not in the document]
Comment (Slavik191) [reply features/cms/customWidgets/widgets/svg/svg.tsx]: done

File: features/cms/customWidgets/widgets/uuid/uuid.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Jabher) [reply features/cms/customWidgets/widgets/uuid/uuid.tsx:72]: future updates may make it dynamic

[the remark this answers is not in the document]
Comment (Slavik191) [reply features/cms/customWidgets/widgets/uuid/uuid.tsx]: done

**Review by Jabher — APPROVED** [review]

**Pull request merged**